apps/internal/base/internal/storage/items.go

```diff
@@ -103,8 +103,14 @@ func (a AccessToken) Key() string {
 	)
 }
 
+// FakeValidate enables tests to fake access token validation
+var FakeValidate func(AccessToken) error
+
 // Validate validates that this AccessToken can be used.
 func (a AccessToken) Validate() error {
+	if FakeValidate != nil {
```

Comment:
It looks like a bad idea. It should be possible to just pass a valid token in the test.

Reply:
To give some context, the cache calls this method before writing a token and won't write that token if this method returns an error. This method will return an error if the token expires within the next 5 minutes. A test that requires the cache to contain an expired token therefore can't pass in a valid token unless it then waits for that token to expire. Perhaps the cache shouldn't check expiry before writing tokens (too big a change to contemplate here) but there are of course other ways to work around this. The test could cache a token that expires in 5 minutes and 1 millisecond then `time.Sleep(2*time.Millisecond)`. I like that approach, but I still prefer `FakeValidate` because it may prove useful in other test cases and doesn't require the one I'm adding in this PR to take a dependency on timing or behavior not under test.